src/python/library/tritonclient/grpc/_infer_stream.py

```diff
@@ -57,18 +57,27 @@ def __init__(self, callback, verbose):
         self._verbose = verbose
         self._request_queue = queue.Queue()
         self._handler = None
+        self._cancelled = False
         self._active = True
+        self._response_iterator = None
 
     def __del__(self):
         self.close()
```

Comment:
I'd advocate for __del__ to be non blocking and use self.close(cancel_requests=True) - unless I'm missing something / good reason to keep it blocking.